### PR TITLE
Add ActivityIndicatorFactory

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: TradeItIosTicketSDK2/ActivityIndicatorFactory.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+@objc public protocol ActivityIndicatorFactory {
+    func build(frame: CGRect) -> UIView
+}
+
+class DefaultActivityIndicatorFactory: ActivityIndicatorFactory {
+    func build(frame: CGRect) -> UIView {
+        let view = UIActivityIndicatorView(frame: frame)
+        view.color = UIColor.gray
+        view.startAnimating()
+        return view
```
Comment (jsom) [inline TradeItIosTicketSDK2/ActivityIndicatorFactory.swift]: `UIActivityIndicatorView` doesn't adjust the spinner size based on what frame you pass in so it just shows a small spinner anyway.

File: TradeItIosTicketSDK2/TradeItSDK.swift [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+    // MARK: Settable properties
+
+    public static var theme: TradeItTheme = TradeItTheme.light()
+    public static var isPortfolioEnabled = true
+    public static var isAdServiceEnabled = false
+    public static var userCountryCode: String? // CountryCode matching standard: ISO3166 alpha-2. Used for managing broker availability.
+    public static var adService: AdService = DefaultAdService()
+    public static var welcomeScreenHeadlineText: String = "Link your broker account to enable:"
+    public static var featuredBrokerLabelText: String = "SPONSORED BROKER"
+    public static var activityViewFactory: ActivityIndicatorFactory = DefaultActivityIndicatorFactory()
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItSDK.swift]: Should I add this to the config calls?

Comment (guillaumedebavelaere) [reply TradeItIosTicketSDK2/TradeItSDK.swift]: If we add this to the trade it screens, yes, otherwise it could be confusing ?

**Comment by guillaumedebavelaere** [comment]
There is also an authenticateCall in the ticket screen where we potentially show a spinner and when we preview and place the order, do we want to use it there ?

**Comment by jsom** [comment]
Still waiting to hear back from Yahoo about `MBProgressHud` spinner

**Comment by ihuxley** [comment]
LGTM, thanks!

**Comment by jsom** [comment]
Merging and we can revisit other spinners later if needed.

**Pull request merged**